### DISCUSSION TITLE
آموزش تانل زدن به روش GOST (GO Simple Tunnel)

### DISCUSSION THREAD

**Discussion by MehdiZakhire** [hiddify/Hiddify-Manager#493] [Q&A سوالات رایج] [answered]
What is the best alternative of ip forwarding tunnels ?
بهترین راه‌کار جایگزین برای آیپی فروارد چیست؟

**Comment by APT-ZERO** [comment]


**Reply by maliohammad** [reply]
> What is the best alternative of ip forwarding tunnels via iptables?

Rinted هم هست

**Comment by MehdiZakhire** [comment]
در مورد سوال اولت بله میشه هم UDP و هم TCP رو تانل کرد.. شما کافیه به جای  TCP بزنی UDP. مثال:

-L=udp://:443/domain.com:443

سوال دومت هم بله وب سوکت روی TCP کار میکنه

سوال سومت هم بله grpc روی TCP کار میکنه
سوال آخرت رو تست نکردم ولی به نظرم نمیشه.. شما تست کردی همینجا اعلام کن

امیدوارم کمک کرده باشم

**Comment by lymanjre** [comment, marked as answer]
# جهت مشاهده مقاله آپدیت شد در خصوص این روش [اینجا](https://hiddify.com/fa/manager/domain-worker-cdn-and-tunneling/Tutorial-for-setting-up-GOST-tunnel-as-relay-server/) را ببینید.

**Reply by salaryy** [reply]
تمام پروتکول ها بستس عزیز هیچ تونلی کار نمیکنه فقط cdn

الان من یک سرور ایران دارم بلا استفاده مونده

**Reply by MehdiZakhire** [reply]
کی همچین چیزی گفته ؟ شما این تانل رو تست کردی؟ عزیزم این تانل IP Forwarding نیست.!!!! این تانل فرق داره نوشتم بالا !! اسمش GOST هست

**Reply by shuvvan** [reply]
سلام و تشکر بابت آموزشتون من قبلا گاست رو اجرا کرده بودم ولی دستوراتش اینجوری نبودن !
این روشی که شما معرفی کردین خیلی عالی و ساده هستش احرا کردم دمت گرم !

**Reply by MehdiZakhire** [reply]
> سلام و تشکر بابت آموزشتون من قبلا گاست رو اجرا کرده بودم ولی دستوراتش اینجوری نبودن ! این روشی که شما معرفی کردین خیلی عالی و ساده هستش احرا کردم دمت گرم !

قربونت امیدوارم مفید باشه و کارت راه افتاده باشه

**Reply by aboly8000** [reply]
سلام
در قسمت domain.com باید یه دامین بزاریم که به سرور خارج اشاره کنه و دایرکت هم باشه؟
در پنل هیدیفای چیکار کنیم؟ همون دامین رو بصورت ریلای ثبت کنیم یا باید یه دامنه دیگه بزاریم که به سرور ایران اشاره کنه؟لطفا ست کردن دامنه ها رو کامل توضیح بدید و بگید مستقیم باشه یا cdn.

**Reply by aboly8000** [reply]
اگه آی پی خارج فیلتر بود چکار کنیم؟

**Reply by hntcAir** [reply]
سلام ممنون از اطلاع و راهنمایی.

آیا میشه روی یک سرور با این روش GOST ، برای چند تا سرور تانل جداگانه ایجاد کرد؟

سپاس از شما

**Reply by vahidabbasi77** [reply]
> در قسمت domain.com باید یه دامین بزاریم که به سرور خارج اشاره کنه و دایرکت هم باشه؟
> در پنل هیدیفای چیکار کنیم؟ همون دامین رو بصورت ریلای ثبت کنیم یا باید یه دامنه دیگه بزاریم که به سرور ایران اشاره کنه؟لطفا ست کردن دامنه ها رو کامل توضیح بدید و بگید مستقیم باشه یا cdn.

**Reply by aboly8000** [reply]
اگه دامنه سرور اصلی رو پشت cdn بزاریم ، این روش کار می کنه؟

**Reply by AnotherAH** [reply]
![image](https://user-images.githubusercontent.com/70831339/231443605-d1580f3a-6067-4022-8ae4-1cc8c0743a82.png)
با یوزر روت هستم و دقیقا مثل شما جلو رفتم

**Reply by ramtin18** [reply]
اگه آی پی خارج فیلتر بود چکار کنیم؟

**Reply by ramtin18** [reply]
اگه دامنه سرور اصلی رو پشت cdn بزاریم ، این روش کار می کنه؟

**Reply by arsaces7** [reply]
> اگه دامنه سرور اصلی رو پشت cdn بزاریم ، این روش کار می کنه؟

اگر فهمیدی به ما هم بگو لطفاً

**Reply by sajadzi** [reply]
> > What is the best alternative of ip forwarding with iptables?
> > بهترین راه‌کار جایگزین برای آیپی فروارد با iptables چیست؟
> 
> یکی از بهترین روشها برای جایگزین کردن ip forwarding با استفاده از iptables استفاده از تانل GOST است.
> 
> این تانل از چند لحاظ خیلی بهتر از IP Forwarding هست. ( یا همان تانل زدن معمولی که همه استفاده میکنند) تو این روش بر خلاف آیپی فورواردینگ شما میتونید به جای IP از نام دامین یا ساب دامین استفاده کنید که این خودش باعث کمتر شناسایی شدن IP سرور شما میشه. یه خوبی دیگه که این روش داره اینه که از یه تانل رمز نگاری شده دیتا عبور میکنه. این تانل Open Source هست. با پروتکل های V2Ray - VMess - Vless - Trojan به خوبی سازگاری داره. در این تانل هم WebSocket جواب میده هم gRPC این تانل چون خودش یک سرویس داره با ریبوت شدن سرور اجرا میشه و نیازی به زدن دستور مجدد نداره. اگر بخواید میتونید هم پورت TCP رو فوروارد کنید و هم UDP. کافیه داخل دستور به جای tcp بزنید udp. میتونید هر چند تا پورت که خواستید رو فوروارد کنید
> 
> ## برای نصب در حالت Relay طبق این آموزش جلو برید.
> وارد سرور ایران خودتون بشید ابتدا سورس برنامه مورد نظر را با کد زیر دانلود و از حالت فشرده خارج کنید.
> 
> ```
> sudo apt install wget nano -y && wget https://github.com/ginuerzh/gost/releases/download/v2.11.5/gost-linux-amd64-2.11.5.gz && gunzip gost-linux-amd64-2.11.5.gz
> ```
> 
> سپس فایلهای آن را به پوشه اجرایی منتقل کنید و به آن دسترسی اجرا دهید.
> 
> ```
> sudo mv gost-linux-amd64-2.11.5 /usr/local/bin/gost && sudo chmod +x /usr/local/bin/gost
> ```
> 
> حالا باید سرویس آن را نیز ایجاد کنید. ابتدا با استفاده از ویرایشگر nano فایل سرویس را ایجاد کنید.
> 
> ```
> sudo nano /usr/lib/systemd/system/gost.service
> ```
> 
> سپس کد زیر را در فایل باز شد به صورت کامل کپی کنید.
> 
> ```
> [Unit]
> Description=GO Simple Tunnel
> After=network.target
> Wants=network.target
> 
> [Service]
> Type=simple
> ExecStart=/usr/local/bin/gost -L=tcp://:443/domain.com:443 -L=tcp://:80/domain.com:80
> 
> [Install]
> WantedBy=multi-user.target
> ```
> 
> بعد از کپی باید نام دامین خودتان را به جای domain.com قرار دهید. توجه کنید که نام دامنه ای که وارد میکنید باید آدرس IP اصلی سرور شما باشد. ( هر ۲ تا domain.com را پاک کنید و نام دامنه خودتون را وارد کنید)
> 
> خوب کار تقریبا تمام شد. حالا با کد زیر ابتدا سرویس gost را استارت کنید. و با سپس سرویس را فعال می‌کنید که با هر بار ریستارت سرویس اجرا شود:
> 
> ```
> sudo systemctl start gost && systemctl enable gost
> ```
> 
> کار تمام شد و میتوانید از تانل استفاده کنید.
> 
> [سورس پروژه GOST](https://github.com/ginuerzh/gost)

اقا این روش ایراد داره رو پنل ssl بهت نمیده و tls از کار میفته

**Reply by hatinati2** [reply]
این سه تا دستور به ترتیب بزنید
systemctl daemon-reload
systemctl restart gost
systemctl status gost

**Reply by masoud7696** [reply]
امتحان کردم. دستور اول دانلود میشه ولی دستور دوم که وارد میکنی خطا میزنه و میگه فایل داخل دایرکتوری وجود نداره

**Reply by etabkr** [reply]
با سلام خدمت استاتید و دوستان محترم
من یه سوال / ابهامی دارم
روی سرور A هیدیفای نصب هست - یه سرور ضعیف B تهیه و این دستورات و اقدامات رو روش انجام دادم
الان سرویس gost روز سرور B نصبه و وضعیتش فعاله
بعدش رو دقیقا متوجه نشدم که الان چه اتفاقی میافته؟ و من چکار باید انجام بدم
- بایت دامنه / ساب دامنه که در فایل gost.service گذاشتم رو برم تو سرور A  (سرور اصلی) توی هیدیفای اضافه کنم و اگر اینطوریه آیا در زمان تعریف در هیدیفای باید آی پی سرور B (سرور واسط ایران) رو بذارم به عنوان آی پی اجباری یا آی پی همون سرور اصلی رو بذارم و حالت دامنه چی باید باشه:
cdn, relay
- آیا باید زیر دامنه رو در کلودفلر به آی پی سرور اصلی متصل کنم یا سرور واسط؟
- من یک دامنه و آی پی سرور اصلی مو به خاطر اینکه از روز اول فکر نمیکردم به این زودی سوخت بشه از دست دادم و فیلتر شدن تو شبکه های موبایلی اکثرا و میخوام با این روش ببینم میشه از سرور استفاده کنم هنوز یا نه
ممنون میشم راهنمایی کنید

**Reply by lymanjre** [reply]
> با سلام خدمت استاتید و دوستان محترم
> 
> من یه سوال / ابهامی دارم
> 
> روی سرور A هیدیفای نصب هست - یه سرور ضعیف B تهیه و این دستورات و اقدامات رو روش انجام دادم
> 
> الان سرویس gost روز سرور B نصبه و وضعیتش فعاله
> 
> بعدش رو دقیقا متوجه نشدم که الان چه اتفاقی میافته؟ و من چکار باید انجام بدم
> 
> - بایت دامنه / ساب دامنه که در فایل gost.service گذاشتم رو برم تو سرور A  (سرور اصلی) توی هیدیفای اضافه کنم و اگر اینطوریه آیا در زمان تعریف در هیدیفای باید آی پی سرور B (سرور واسط ایران) رو بذارم به عنوان آی پی اجباری یا آی پی همون سرور اصلی رو بذارم و حالت دامنه چی باید باشه:
> 
> cdn, relay
> 
> - آیا باید زیر دامنه رو در کلودفلر به آی پی سرور اصلی متصل کنم یا سرور واسط؟
> 
> - من یک دامنه و آی پی سرور اصلی مو به خاطر اینکه از روز اول فکر نمیکردم به این زودی سوخت بشه از دست دادم و فیلتر شدن تو شبکه های موبایلی اکثرا و میخوام با این روش ببینم میشه از سرور استفاده کنم هنوز یا نه
> 
> ممنون میشم راهنمایی کنید
> 
> 

سلام تکمیلش میکنیم از ابهام خارج شه

باید ریلی ثبت کنید. یه سرچ بزن ws tunnel را ببین قسمت اخرشو مث اون برو جلو

**Reply by esmaeilifardi** [reply]
> این سه تا دستور به ترتیب بزنید systemctl daemon-reload systemctl restart gost systemctl status gost

برا حل این مشکل در سرور ایران دامین بندازین پشت nginx و cerificat را هم یا از کلودفلر بزنین یا certbot و در nginx کانفیگ پورت 443 بزارین

**Reply by masiha148** [reply]
دوستان کسی می‌تونه بهم کمک کنه در مورد تانل زدن؟

**Comment by Shahradazar** [comment]
امکان داره یه فیلم آموزشی تهییه بکنید هزینش پرداخت میشه

**Comment by mafayyaz** [comment]
من این روش رو اجرا کردم و کار هم می‌کنه، خیلی ممنونم.
ولی مشکلی که هست اینه که آپلود خوبی نداره و پینگش مثل CDN داخلی خوب نیست. در نتیجه توی تماس اینترنتی عملکرد خوبی نداره و خیلی به درد من نخورد.

**Reply by MehdiZakhire** [reply]
دوست من این روش فقط تانل هست !!!! این که سرعت دانلود آپلود سرور ایران شما پایین باشه با این روش بالا نمیره سرعت... 
شما باید یه سرور پیدا کنید که دانلود و آپلود به خارج روش بالا باشه که خیلی هم سخت گیر میاد...
من و چندین نفر داریم از این تانل استفاده میکنیم و سرعت دانلود روی ایرانسل تا 215 مگ و آپلود هم تا 80 بوده

**Reply by mafayyaz** [reply]
من چند روزه دارم از آی پی تمیز استفاده می کنم توی سرور ایران و واقعا تفاوت محسوس شده و کیفیت عالی.
جایی که من ازش سرویس می‌گیرم، دایرکتش جالب نیست ولی با CDN و آی پی تمیز تقریبا کل پهنای باندم رو می گیرم.
واقعا دم شما گرم.

**Comment by robin98** [comment]
من متوجه شدم که این اسکریپت، نسخه داکر هم داره؛
آیا روشی وجود داره که بشه این مدل تانلینگ رو مثل `hpello/tcp-proxy` روی سرویس های PaaS (مثل هم‌روش) اجرا کرد؟
[منظورم این روش هست](https://github.com/hiddify/hiddify-config/discussions/129#discussioncomment-4996908)

**Comment by talamag** [comment]


**Reply by mafayyaz** [reply]
توی همون فایل gost.service، از این دستور استفاده کنید:
-L=tcp://:8888/domain.com:9999
۸۸۸۸ پورت ورودی به سرور ایران و ۹۹۹۹ پورتی هست که می خواید به سرور خارج دیتا ارسال کنید.
بعد از تغییرات هم این دستورات رو بزنید:
sudo systemctl daemon-reload

sudo systemctl stop gost && systemctl disable gost

sudo systemctl start gost && systemctl enable gost

**Reply by mafayyaz** [reply]
هر دو پورت ورود و خروج ۲۰۵۳ هست؟

**Comment by Shahradazar** [comment]
داداشم میشه توی تلگرام به این آیدی یه پیام بدید
@robyshop1

**Comment by hntcAir** [comment]
از یک سرور در ایران میشه همزمان استفاده کرد با GOST به دو تا سرور جداگانه خارجی زد با پنل هیدیفای؟

**Reply by hntcAir** [reply]
پاسخ به خودم
بله میشه و این کار رو ست کردم.


اما پنل هیدیفای مشکل درج پورت در بخش‌های مختلف از جمله URLی که برای Subs تولید می‌کند دارد.

**Reply by masoumi-pezhman** [reply]
سلام چطور این مورد رو انجام دادی؟ میشه راهنمایی کنی ؟ 
منم میخوام همینکار رو انجام بدم

**Reply by ParixG** [reply]
سلام چطور میشه این کار رو کرد؟

**Comment by pythondeveloper123** [comment]
با سلام و سپاس از آموزش خوبتون
من تا قسمت آخر پیش می روم و کد رو کپی و نام دامنه ها رو ست می کنم اما 2 تا مشکل دارم:

1. بعد از save کردن مرحله آخر ارور:  [error writing /usr/bin/systemd/system/gost.service: no such file or directory]     دریافت می کنم.

2. در قسمت دامنه ها، ساب دامین رایگانی که در سایت afraid.org  ثبت کردم را وارد نمودم، آیا این صحیح است؟

سپاس از حضور گرمتون.

**Comment by onyxtm** [comment]
امکانی هست که به هر پورتی تو سرور ایران درخواست داده میشه مثل ای پی تیبل همون رو منتقل کرد به سرور مقصد؟

**Comment by sajadzi** [comment]
دوستان من گاست رو راه انداختم، ایپیه متصل به دامینم هم پشت cdn کلافلره.ایپیم فیلتره کارم میکنه سرعت دانلود حرف نداره ولی اپلود یکم ضعیفه.اونم بخاطر cdn کلاده که ایپیش سرعتش کمه .میخواستم بدونم میتونم یکاری کنم اون ایپه cdn به صورت اجباری تغییر کنه مثل force cdn پنل هیدیفای؟

**Comment by johns1976** [comment]
آقا در مورد سرور ایران فقط نوشتید روی سرور خارج هیچکاری نمیخواد بکنیم ؟
اتومات کانفیگ میشه؟

**Comment by shahram-noorani** [comment]
سلام لطفا اموزش استفاده ازgost روی centos یا debian رو بزارید

**Comment by aminkg** [comment]
سلام این روش تانلینگ پشت پروکسی cloudflare کار نمیکنه؟ حتما باید ساب دامینمون آی پی خود سرور رو برگردونه؟

**Comment by rnj79** [comment]
اگر سروری که میخوایم ازش تونل بزنیم . دو تا آیپی داشته باشه چطور باید کد بزنیم . 
eth0 یک آیپی 
eth1 یک آیپی دیگه هست

**Comment by Hooti66** [comment]
سلام دوستان من مدتی هستش که هر ساب دامنه ای میذارم بعد از چند ساعت فیلتر میشه روی چندتا پنل هدیفای دیگه امتحان کردم برای دوستانم فیلتر نمیشه و خیلی راحت کار میکنه بدون فیلتر ولی روی هدیفای خودم هر چند ساعت فیلتر میشه ساب دامنه هام 
پنل های هدیفای بقیه رو هم خودم نصب کردم عین همدیگه هستند ولی برای من تند تند میره 
کسی دلیلش رو میدونه ؟

**Comment by masoumi-pezhman** [comment]
دامنه رو عوض کن کلا

**Comment by masoudgb** [comment]
دوستان سلام
اگر ممکنه کامنت سمت سرور و کلاینت رو برای استفاده از tls و sni بذارید با سپاس

**Comment by Blaster-777** [comment]
از نظر تعداد یوزر محدودیتی داره؟ مثلا روی یک پورت حدودا چند تا یوزر میشه داشته باشیم؟ دیدم بعضی ها میگن تا 50 تا یوزر روی یک پورت خوب جواب میده بیشتر بشه افت سرعت شدیدی داره کسی امتحان کرده؟

**Reply by hope-s** [reply]
تاثیر خاصی نداره

**Comment by yadegari1** [comment]
دوستان خواهشا  کمک کنید میخوام پورت فورواردینگ بکنم برای اپلیکیشن موبایل.هزینه هم میدم کسی راهنمایی بکنه.ایمیل بزنید صحبت کنیم
yadegari20002000@gmail.com
